src/gleam/httpc.gleam

```diff
@@ -98,6 +145,124 @@ pub fn send_bits(
+    False -> [Ssl([Verify(VerifyNone)]), ..erl_http_options]
+  }
+  let erl_options = [
+    BodyFormat(Binary),
+    SocketOpts([Ipfamily(Inet6fb4)]),
+    Sync(False),
+    Stream(#(Self, Once)),
+  ]
+  use request_id <- result.try(
+    case req.method {
+      http.Options | http.Head | http.Get -> {
+        let erl_req = #(erl_url, erl_headers)
+        erl_stream_request_no_body(
+          req.method,
+          erl_req,
+          erl_http_options,
+          erl_options,
+        )
+      }
+      _ -> {
+        let erl_content_type =
+          req
+          |> request.get_header("content-type")
+          |> result.unwrap("application/octet-stream")
+          |> charlist.from_string
+        let erl_req = #(erl_url, erl_headers, erl_content_type, req.body)
+        erl_stream_request(req.method, erl_req, erl_http_options, erl_options)
+      }
+    }
+    |> result.map_error(normalise_error),
+  )
+
+  Ok(request_id)
+}
+
+/// Triggers the next asynchronous streaming message to be sent to the calling process
+/// designated by `pid`. 
+/// 
+@external(erlang, "gleam_httpc_ffi", "receive_next_stream_message")
+pub fn receive_next_stream_message(id: process.Pid) -> Nil
+
+@external(erlang, "gleam_httpc_ffi", "coerce_stream_message")
+fn decode_stream_message(msg: Dynamic) -> RawStreamMessage
+
+/// Configure a selector to receive stream messages
+/// 
+/// Note this will receive messages from all processes that sent a HTTP stream request;
+/// for example using `send_stream_request`, rather than any specific one.
+/// In this case, for finer grained processing, you can filter on the `RequestIdentifier`,
+/// which is the first argument in the `StreamMessage` constructor.
+/// If you wish to only handle stream messages from one process, then use one
+/// process per HTTP stream request. 
+///
+/// ## Example
+///
+/// ```gleam
+/// process.new_selector() |> select_stream_messages(raw_stream_mapper())
+/// ```
+/// 
+pub fn select_stream_messages(
+  selector: process.Selector(t),
+  mapper: fn(RawStreamMessage) -> t,
+) -> process.Selector(t) {
+  let http = atom.create(http.scheme_to_string(http.Http))
+  let map_stream_message = fn(mapper) {
+    fn(message) { mapper(decode_stream_message(message)) }
+  }
+
+  selector
+  |> process.select_record(http, 1, map_stream_message(mapper))
+}
+
+/// Converts a raw stream message into a user-facing `StreamMessage`.
+///
+/// This mapper is primarily used to transform header values from
+/// `List(#(Charlist, Charlist))` into the more idiomatic `List(#(String, String))`,
+/// which is easier to work with in Gleam.
+///
+/// You can use this function as the `mapper` argument to `select_stream_messages/2`,
+/// or you can supply your own custom mapper if you need additional transformations.
+///
+/// ## Example
+///
+/// ```gleam
+/// process.new_selector() |> select_stream_messages(raw_stream_mapper())
+/// ```
+///
+pub fn raw_stream_mapper() -> fn(RawStreamMessage) -> StreamMessage {
```

Comment:
This looks like part of the internal implementation and shouldn't be part of the public API 🙏